detox/android/detox/src/full/java/com/wix/detox/espresso/DetoxAction.java

```diff
@@ -149,6 +152,11 @@ public static ViewAction scrollToIndex(int index) {
         return new ScrollToIndexAction(index);
     }
 
+    public static ViewAction setDatePickerDate(String dateString) {
+        ZonedDateTime date = ZonedDateTime.parse(dateString);
+        return PickerActions.setDate(date.getYear(), date.getMonthValue(), date.getDayOfMonth());
+    }
```

Comment:
As I said in the previous comment, I suggest to add a `dateFormat` / `pattern` parameter and to support the other case:
```java
ZonedDateTime date;
if (pattern.equals("ISO8601")) {
    date = ZonedDateTime.parse(dateString);
} else {
    DateTimeFormatter formatter = DateTimeFormatter.ofPattern(pattern);
    date = ZonedDateTime.parse(dateString, formatter);
}

return PickerActions.setDate(date.getYear(), date.getMonthValue(), date.getDayOfMonth());
```

Reply:
Thanks. I implemented this. I had to use LocalDate though for the pattern case as ZonedDateTime complains if it does not get a time or timezone.

Reply:
As mentioned in another thread. This works now and is implemented with a simpledateformatter now.